### PR TITLE
CTT-149 version as git tag, only allow pip install

### PR DESCRIPTION
**Pull request by andylytical**
I propose this is a minor version change, from 1.8.x -> 1.9.x, since old installation methods have been removed and docs supporting old install methods are either gone or moved aside.

for 1.9.0
* Pip is now the only supported install method
* Simplified install to a handful of copy-n-paste instructions (config files require manual edits but they are simplified to just one or two edits in the most common case)
* Modernize packaging with pyproject.toml, which enables:
  * Version checks, requirements, and dependencies managed automatically by pip
  * Version is now tracked using git tags
* ca_certs updated to include only Let'sEncrypt and InCommon root CA certs
* Add a script to manage (rebuilding or appending certs to) the ca_certs file
* Add github actions to publish to pypi and test.pypi
* Add support to install dev versions from test.pypi
* Old docs moved to "xsede" folder pending further review (see README in that folder)
* In-progress docs moved to "drafts" folder

### CODE REVIEW

**Comment by andylytical** [comment]
The errors mentioned (all related to unsuccessful deployment) are due to setuptools version numbering using git tags and now that I've rebased, setuptools will "generate" a dev version that's been published before and cannot ever reuse a version number once published to test.pypi (same with pypi, but we'll only ever publish there on a git tag, which is only when a PR is merged).

File: docs/history.md [hunk 1]
```diff
@@ -1,38 +1,10 @@
...
-
-## License
-
-This software is licensed the Apache License Version 2.0.
-
-
-## Acknowledgements
+Workflow steps specify what Data they require and what Data they produce. This allows IPF to construct workflows based on partial information - in the case where there are not steps that produce the same Data, an entire workflow can be constructed from a single publish step and its required input Data. At the other extreme, workflows can be exactly specified with specific steps identified and the outputs of steps bound to the inputs of other steps. A typical workflow (e.g. GLUE 2) specifies what steps to include but lets IPF automatically link outputs to inputs of these steps.
 
-This work was supported by the TeraGrid, XSEDE, FutureGrid, XSEDE 2, and ACCESS CONECT projects under
```
Comment (ericblau) [inline docs/history.md:37]: We should probably keep this supported by statement somewhere.

File: docs/install.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Install
+1. Get the setup script
+    ```
+    curl -o ~/ipf-setup.sh https://raw.githubusercontent.com/access-ci-org/ipf/refs/heads/CTT-149/fix_version/setup.sh
```
Comment (ericblau) [inline docs/install.md]: We probably shouldn't be referencing branches for production documentation

**Review by ericblau — COMMENTED** [review]
I think in general this is great.  I like the approach of using Pip, and automatically publishing to PyPi more than bespoke installers from github.  I am slightly concerned about the failed check on this PR--is it known why the upload to test.pypi.org failed for this PR?  Not concerned about the approach, mind you, just if there's an edge case not accounted for in the publishing

File: docs/history.md (hunk 1 above)
Comment (andylytical) [reply docs/history.md:37]: The supported by statement is on the main README. I removed it from the "history.md" because it seems redundant and of the two locations, I think the README is more prominent.

File: docs/install.md (hunk 2 above)
Comment (andylytical) [reply docs/install.md]: Good catch. I fixed this in the latest commit.

**Comment by andylytical** [comment]
Description of the failed publish to test.pypi

## Problem
Cannot (ever) re-publish a specific version, even if first deleted from pypi. Setuptools_scm will auto-generate a previously used version string after a rebase.

## Resolution
Ensure "dev" versions are always unique.

I spent some time last weekend researching if setuptools_scm can do the above and found no (easy) way. I propose investigating other build tools to see if one has such a capability. Now that we are using pyproject.toml and an automated build & publish workflow, I suspect using a different build tool will be trivial and will not have a negative impact on the user experience. That work can be done independently of this PR #23 and should be done in a separate ticketed effort.

File: docs/history.md (hunk 1 above)
Comment (andylytical) [reply docs/history.md:37]: From Slack, Eric said ready to merge. Marking this conversation resolved.

File: docs/install.md (hunk 2 above)
Comment (andylytical) [reply docs/install.md]: from Slack, Eric said ready to merge. Marking this conversation resolved.

**Comment by andylytical** [comment]
For posterity sake, noting here that the failed publish to test.pypi is not limited to just the "rebase" workflow. It also occurs with the "squash and merge" workflow. For an example, see https://github.com/andylytica/ajlpypubtest.

There, I created a branch, made two commits, both of which published to test.pypi. Then I created a PR and merged using "squash and merge", which failed to publish to test.pypi. This is expected if we think about what happens ... the resulting history after squash and merge looks identical to rebase, namely, a new commit is created whose direct parent is "main" (allowing a FF merge, thus a linear history) and therefore setuptools_scm generates a previously used tag (specifically, x.y.z.dev1). Published version strings can never be reused, thus the publish step fails.

Note then that the "Release" process does create a new tag and at that point publishing to test.pypi and pypi both succeed and with the new tagged version (without a dev suffix).
